hasher-matcher-actioner/hmalib/common/label_models.py

```diff
@@ -0,0 +1,82 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+import hmalib.common.config as config
+import json
+import typing as t
+
+from dataclasses import dataclass, field, fields
+from hmalib.common.logging import get_logger
+from requests import get, post, put, delete, Response
+
+logger = get_logger(__name__)
+
+
+@dataclass(unsafe_hash=True)
+class Label:
+    key: str
+    value: str
+
+    def to_dynamodb_dict(self) -> dict:
+        return {"K": self.key, "V": self.value}
+
+    @classmethod
+    def from_dynamodb_dict(cls, d: dict):
+        return cls(d["K"], d["V"])
+
+    def __eq__(self, another_label: object) -> bool:
+        if not isinstance(another_label, Label):
+            return NotImplemented
+        return self.key == another_label.key and self.value == another_label.value
+
+
+@dataclass(unsafe_hash=True)
+class ClassificationLabel(Label):
+    key: str = field(default="Classification", init=False)
+
+
+@dataclass(unsafe_hash=True)
+class BankSourceClassificationLabel(ClassificationLabel):
+    key: str = field(default="BankSourceClassification", init=False)
+
+
+@dataclass(unsafe_hash=True)
+class BankIDClassificationLabel(ClassificationLabel):
+    key: str = field(default="BankIDClassification", init=False)
+
+
+@dataclass(unsafe_hash=True)
+class BankedContentIDClassificationLabel(ClassificationLabel):
+    key: str = field(default="BankedContentIDClassification", init=False)
+
+
+@dataclass(unsafe_hash=True)
+class ActionLabel(Label):
+    key: str = field(default="Action", init=False)
+
+
+@dataclass(unsafe_hash=True)
+class WritebackLabel(Label):
```

Comment:
Hmm, @eriksendc, this seems to be doubling down on action labels, which I wish were not the case. I've gotten to the point where I regret the creation of the Label class :( I don't want to live in a world where everything is a label.

Reply:
Talked with @eriksendc offline, and the plan is to eventually refactor off label (either just rename or structure heirarchy change), my concerns are addressed!